mteb/models/google_models.py

```diff
@@ -0,0 +1,102 @@
+"""Required python >=3.9 and that the `google-generativeai` package is installed. Additionally the GOOGLE_API_KEY environment variable must be set."""
+
+from __future__ import annotations
+
+from functools import partial
+from typing import Any
+
+import numpy as np
+
+from mteb.encoder_interface import Encoder
+from mteb.model_meta import ModelMeta
+
+
+class GoogleTextEmbeddingModel(Encoder):
+    def __init__(self, model_name: str, sep: str = " ", **kwargs) -> None:
+        self.model_name = model_name
+
+    def _embed(
+        self, sentences: list[str], *, task_type: str, titles: list[str] | None = None
+    ) -> np.ndarray:
+        try:
+            import google.generativeai as genai
+        except ImportError:
+            raise ImportError(
+                "`google-generativeai` is required to run the google API, pleae install it using `pip install google-generativeai`"
+            )
+
+        if titles:
+            result = genai.embed_content(  # type: ignore
+                model=self.model_name,
+                content=sentences,
+                task_type=task_type,
+                title=titles,
+            )
+        else:
+            result = genai.embed_content(  # type: ignore
+                model=self.model_name,
+                content=sentences,
+                task_type=task_type,
+            )
+
+        return np.asarray(result["embedding"])
+
+    def encode(
+        self,
+        sentences: list[str],
+        prompt_name: str | None = None,
+        **kwargs: Any,
+    ) -> np.ndarray:
+        input_type = "RETRIEVAL_DOCUMENT"  # Default
```

Comment:
Since this one seems to be optional (https://cloud.google.com/vertex-ai/generative-ai/docs/model-reference/text-embeddings-api#request_body), maybe it should default to None? I.e. for PairCLF we'd rather use None than `RETRIEVAL_DOCUMENT`, no?